### ISSUE TITLE
MD035 is triggered by horizontal line in list

### ISSUE THREAD

**Issue by NomarCub** [DavidAnson/markdownlint#736]
`hr-style` triggers for consistent styles that are inside a list. Here's an [example](https://spec.commonmark.org/dingus/?text=-%20a%0A%20%20-%20b%0A%20%20-%20---%0A%20%20-%20c%0A-%20d%0A-%20---%0A-%20e) of valid markdown:

```md
- a
  - b
  - ---
  - c
- d
- ---
- e
```

**Comment by nschonni** [comment]
You'll notice that the HRs actually breakup the list and aren't treated as part of the list

```
- a
  - b
---
  - c
- d
---
- e
```

**Comment by DavidAnson** [comment]
The sample as presented does not produce MD035 warnings for me. However, adding another horizontal rule to the top of the sample with the same style **does** produce a warning which I agree is inappropriate: https://dlaa.me/markdownlint/#%25m%23%20Issue%20736%0A%0A---%0A%0A-%20a%0A%20%20-%20b%0A%20%20-%20---%0A%20%20-%20c%0A-%20d%0A-%20---%0A-%20e%0A

**Comment by DavidAnson** [comment]
It sounds like your original issue may have been a misunderstanding, but I think there is a legitimate problem revealed by the sample i link to above. I will try to fix that problem. I can either use this issue as a reminder or you can close it if you'd like and I will leave myself a note on my private TODO list.

**Comment by NomarCub** [comment]
Sure, I get the sequence of events, and I'll leave this open. You might want to rename it to reflect the actual problem for others stumbling onto it in the meantime.

And my last question still stands. I had my example text and you posted yours. Should the rule should produce a warning *both* for my text and yours, or for *neither*?

**Comment by DavidAnson** [comment]
The horizontal rule style in your original sample is consistent, so there should be no warnings for that – and there are not. The style is also consistent in the modified version I link to so there should not be warnings there – but there are. That is the bug and once I fix it, neither example will produce a warning about inconsistent horizontal rule style.

**Comment by NomarCub** [comment]
That's what I wanted originally too. But now I see that `- ---` gets interpreted as `---` by CommonMark, so it really does seem like an inconsistent hr. What's the rationale behind not giving a warning?
